### PR TITLE
API Changes: Sol-str-slice #2143

### PR DESCRIPTION
**Pull request by amandawrcoelho**
Replace the caseeq keyword with case_eq

sol_str_slice_to_int() now receive a pointer to long int.

issue #1663

Signed-off-by: Amanda Coelho <amanda.winder.ribeiro.coelho@intel.com>

### CODE REVIEW

File: src/shared/sol-fbp-parser.c [hunk 1]
```diff
@@ -29,7 +29,7 @@ struct sol_fbp_parser {
     struct sol_fbp_scanner scanner;
 
     /* Stores both the current and the pending token, so we can peek
-    * at the next token without having to calculate it every time. */
+     * at the next token without having to calculate it every time. */
     struct sol_fbp_token current_token;
```
Comment (ceolin) [inline src/shared/sol-fbp-parser.c]: Extra space added unecessarily

File: src/shared/sol-fbp-parser.c [hunk 2]
```diff
@@ -255,7 +256,10 @@ parse_inport_stmt(struct sol_fbp_parser *p)
 
     sol_fbp_graph_add_in_port(p->graph, node_idx, port, get_token_position(&p->current_token));
 
-    err = sol_fbp_graph_add_exported_in_port(p->graph, node_idx, port, port_idx, exported_port, get_token_position(&p->current_token), &ep);
+    if ((long)(int)port_idx != port_idx)
+        return false;
```
Comment (ceolin) [inline src/shared/sol-fbp-parser.c]: above you did cast to `long int`, although they are identical, keep it consistent choosing one way to do the cast.

File: src/shared/sol-fbp-parser.c [hunk 3]
```diff
@@ -596,9 +601,15 @@ parse_conn_stmt(struct sol_fbp_parser *p)
 
         sol_fbp_graph_add_in_port(p->graph, dst, dst_port_name, in_port_position);
 
-        r = sol_fbp_graph_add_conn(p->graph, src, src_port_name, src_port_idx, dst, dst_port_name, dst_port_idx, conn_position);
+        if ((long)(int)src_port_idx != src_port_idx)
+            return false;
```
Comment (ceolin) [inline src/shared/sol-fbp-parser.c]: ditto

File: src/shared/sol-fbp-parser.c [hunk 4]
```diff
@@ -596,9 +601,15 @@ parse_conn_stmt(struct sol_fbp_parser *p)
...
+
+        if ((long)(int)dst_port_idx != dst_port_idx)
+            return false;
+
+        r = sol_fbp_graph_add_conn(p->graph, src, src_port_name, (int)src_port_idx, dst, dst_port_name, (int)dst_port_idx, conn_position);
         if (r < 0)
-            return handle_conn_error(p, src, &src_port_name, src_port_idx, dst, &dst_port_name, dst_port_idx, &conn_position, r);
+            return handle_conn_error(p, src, &src_port_name, (int)src_port_idx, dst, &dst_port_name, (int)dst_port_idx, &conn_position, r);
```
Comment (ceolin) [inline src/shared/sol-fbp-parser.c]: Is guaranteed that this won't cause an overflow ?

Comment (amandawrcoelho) [inline src/shared/sol-fbp-parser.c]: Here is the overflow check

+        if ((long)(int)src_port_idx != src_port_idx)
+            return false;

and 

+        if ((long)(int)src_port_idx != src_port_idx)
+            return false;

Comment (amandawrcoelho) [inline src/shared/sol-fbp-parser.c]: @ceolin tks for the review :D

**Pull request closed without merging**